digitaltwin_client/src/dt_interface_client.c

```diff
@@ -1163,6 +1163,7 @@ static void ProcessPropertyUpdateIfNeededFromDesired(DT_INTERFACE_CLIENT* dtInte
             if (dtInterfaceReportedJson)
             {
                 payloadForInitialProperty = GetPayloadFromProperty(dtInterfaceReportedJson, propertyName);
+                json_object_remove(dtInterfaceReportedJson, propertyName);
```

Comment:
>json_object_remove [](start = 16, length = 18)

Could you please put a comment explaining the intent of this.  Namely so that it signals to caller that we have invoked the callback for this reported property and reported-only-properties should be skipped.

I don't see how else to solve this problem more cleanly than this, but we typically don't remove items like this so we should call out why.